posts/self-hosted-homelab/index.md

```diff
@@ -0,0 +1,205 @@
+
+[pic of Marvin and the rest of my homelab]
+
+I installed Debian GNU/Linux and tucked Marvin into the homelab closet. Marvin’s only connection to the outside world is an ethernet cable.
+
+## Install Docker then Coder
+
+I ssh'd into Marvin to get started.
+
+First I installed Docker, before Coder. You'll see why later.
+
+Instead of [Docker Desktop](https://www.docker.com/products/docker-desktop/), I installed [Docker Engine for Debian](https://docs.docker.com/engine/install/debian/) because Marvin is headless. I'm not missing out on Docker Desktop's GUI admin features since Coder does a lot of Docker management for me.
+
+Next was Coder. Its [install script](https://coder.com/docs/v2/latest/install/install.sh) does the right thing. In Marvin’s case it detected Debian to install a `.deb` package. It also recognized Docker so it added the `coder` user to the `docker` group.
+
+```bash
+marc@marvin:~$ sudo curl -fsSL https://coder.com/install.sh | sh
+[sudo] password for marc:
+Debian GNU/Linux 11 (bullseye)
+Installing v0.18.1 of the amd64 deb package from GitHub.
+# Installation progress...
+deb package has been installed.
+# Info for next steps, including setting up Coder as a systemd server...
+marc@marvin:~$
+```
+
+I wanted to dedicate Marvin to hosting my projects, so I followed the installer's suggestion to run Coder [as a service](https://coder.com/docs/v2/latest/admin/configure#system-packages):
+
+```bash
+marc@marvin:~$ sudo systemctl enable --now coder
+```
+
+Accessing Coder is straightforward:
+
+- Command line tool
+- Web interface that I can access on my home network with Marvin's local IP address
+- Publicly accessible, encrypted tunnel based on [Tailscale](https://coder.com/docs/v2/latest/networking)
+
+I got the [tunnel’s access URL](https://coder.com/docs/v2/latest/admin/configure) (and checked that Coder is up and running):
+
+```bash
+marc@marvin:~$ sudo journalctl -u coder.service -b
+# Log entries from Coder
+Mar 09 14:22:21 marvin coder[617]: Opening tunnel so workspaces can connect to your deployment. For production scenarios, specify an external access URL
+Mar 09 14:22:22 marvin coder[617]: View the Web UI: https://marvin-access-url.pit-1.try.coder.app
+# More log entries from Coder
+```
+
+## Using Coder for the first time
+
+At this point I was able to use my laptop's browser to log in to Coder. I opened a tab with the access URL and was asked to set up an account. This account is for Marvin's instance of Coder only.
+
+![Setting up an account](./static/account-setup.png)
+
+Once logged in, I was ready to set up my first workspace. A Coder [workspace](https://coder.com/docs/v2/latest/workspaces) is a runnable environment that a developer, well, works in. Each developer has their own workspace (or workspaces).
```

Comment:
It might make sense to explain further what remote development environments are at this point (at least with regards to Coder), since that term often has a few meanings (e.g. a preview/test environment versus where the actual coding happens).

Many people are also unaware that editors can run in the web browser (e.g. [code-server](https://github.com/coder/code-server)) or even with a local VS Code/JetBrains IDE and remote language server with VS Code Remote SSH or JetBrains gateway.

Maybe this just means we change to 

```suggestion
Once logged in, I was ready to set up my first workspace. A Coder [workspace](https://coder.com/docs/v2/latest/workspaces) is a runnable environment that a developer, well, works in and connects to from their favorite IDE (e.g. VS Code, JetBrains, vim). Each developer has their own workspace (or workspaces).
```

Reply:
Excellent suggestion. Thank you!